packages/backend/src/open_payments/quote/routes.ts

```diff
@@ -0,0 +1,128 @@
+import { Logger } from 'pino'
+import { validateId } from '../../shared/utils'
+import { AppContext } from '../../app'
+import { IAppConfig } from '../../config/app'
+import { QuoteService } from './service'
+import { isQuoteError, errorToCode, errorToMessage } from './errors'
+import { Quote } from './model'
+import { Amount, parseAmount } from '../amount'
+
+interface ServiceDependencies {
+  config: IAppConfig
+  logger: Logger
+  quoteService: QuoteService
+}
+
+export interface QuoteRoutes {
+  get(ctx: AppContext): Promise<void>
+  create(ctx: AppContext): Promise<void>
+}
+
+export function createQuoteRoutes(deps_: ServiceDependencies): QuoteRoutes {
+  const logger = deps_.logger.child({
+    service: 'QuoteRoutes'
+  })
+  const deps = { ...deps_, logger }
+  return {
+    get: (ctx: AppContext) => getQuote(deps, ctx),
+    create: (ctx: AppContext) => createQuote(deps, ctx)
+  }
+}
+
+async function getQuote(
+  deps: ServiceDependencies,
+  ctx: AppContext
+): Promise<void> {
+  const { quoteId: quoteId } = ctx.params
+  ctx.assert(validateId(quoteId), 400, 'invalid id')
+  const acceptJSON = ctx.accepts('application/json')
+  ctx.assert(acceptJSON, 406)
+
+  const quote = await deps.quoteService.get(quoteId)
+  if (!quote) return ctx.throw(404)
+
+  const body = quoteToBody(deps, quote)
+  ctx.body = body
+}
+
+async function createQuote(
```

Comment:
Where do we incorporate the fact that the wallet can change the details of the quote to always make it a fixed-delivery payment? I may have missed it...

Reply:
During quote creation, Rafiki sends the quote to the wallet's `quoteUrl`.
The wallet may increase the `sendAmount` of a fixed-delivery quote or decrease the `receiveAmount` of a fixed-send quote.
At this point
>Quote amounts are fixed, so an outgoing payment is sent as fixed-receive.

Reply:
I see. And why do we not use the webhook service for that?

Reply:
Because we depend on the quote returned in the response body.
And I didn't want to break our (and [others](https://stripe.com/docs/webhooks/quickstart)') convention of webhook requests just expecting a `200` response
https://github.com/interledger/rafiki/blob/51128d2ac1bf63214a8a67a19b4ec8aeea1e93b8/packages/backend/src/webhook/service.ts#L102